napari/_vispy/layers/base.py

```diff
@@ -195,6 +195,13 @@ def _on_matrix_change(self):
         translate = transform.translate[::-1]
         matrix = transform.linear_matrix[::-1, ::-1].T
 
+        if (
+            self._array_like
+            and self.layer._slice_input.ndisplay == 3
+            and self.layer.multiscale
+        ):
+            translate = translate + (self.layer.downsample_factors[-1] - 1) / 2
```

Comment:
`translate` has already been flipped to vispy axis ordering, whereas the downsample factors are still in numpy ordering, so you have to flip them here

```suggestion
            translate = translate + (self.layer.downsample_factors[-1][::-1] - 1) / 2
```

or define `translate` before flipping.